data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v3/GetNewBlockV3IntegrationTest.java

```diff
@@ -0,0 +1,328 @@
+    when(validatorApiChannel.createUnsignedBlock(eq(UInt64.ONE), eq(signature), any()))
+        .thenReturn(SafeFuture.completedFuture(Optional.of(blindedBlockContents)));
+    when(executionLayerBlockProductionManager.getCachedPayloadResult(UInt64.ONE))
+        .thenReturn(
+            Optional.of(
+                new ExecutionPayloadResult(
+                    mock(ExecutionPayloadContext.class),
+                    Optional.empty(),
+                    Optional.empty(),
+                    Optional.empty(),
+                    Optional.of(SafeFuture.completedFuture(executionPayloadValue)))));
+    Response response = get(signature, ContentTypes.JSON);
+    assertThat(response.code()).isEqualTo(SC_OK);
+    assertThat(response.header(HEADER_CONSENSUS_VERSION))
+        .isEqualTo(specMilestone.name().toLowerCase(Locale.ROOT));
+    assertThat(response.header(HEADER_EXECUTION_PAYLOAD_BLINDED)).isEqualTo(Boolean.toString(true));
+    assertThat(response.header(HEADER_EXECUTION_PAYLOAD_VALUE))
+        .isEqualTo(executionPayloadValue.toDecimalString());
+    final String body = response.body().string();
+    assertThat(body).isEqualTo(getExpectedBlockAsJson(specMilestone, true, true));
+  }
+
+  @TestTemplate
+  void shouldGetBlindedBlockContentAsJsonPostDenebAsSsz() throws IOException {
+    assumeThat(specMilestone).isEqualTo(DENEB);
+    final BlindedBlockContents blindedBlockContents =
+        dataStructureUtil.randomBlindedBlockContents(ONE);
+    final BLSSignature signature = blindedBlockContents.getBlock().getBody().getRandaoReveal();
+    when(validatorApiChannel.createUnsignedBlock(eq(UInt64.ONE), eq(signature), any()))
+        .thenReturn(SafeFuture.completedFuture(Optional.of(blindedBlockContents)));
+    when(executionLayerBlockProductionManager.getCachedPayloadResult(UInt64.ONE))
+        .thenReturn(
+            Optional.of(
+                new ExecutionPayloadResult(
+                    mock(ExecutionPayloadContext.class),
+                    Optional.empty(),
+                    Optional.empty(),
+                    Optional.empty(),
+                    Optional.of(SafeFuture.completedFuture(executionPayloadValue)))));
+    Response response = get(signature, ContentTypes.OCTET_STREAM);
+    assertThat(response.code()).isEqualTo(SC_OK);
+    assertThat(response.header(HEADER_CONSENSUS_VERSION))
+        .isEqualTo(specMilestone.name().toLowerCase(Locale.ROOT));
+    assertThat(response.header(HEADER_EXECUTION_PAYLOAD_BLINDED)).isEqualTo(Boolean.toString(true));
+    assertThat(response.header(HEADER_EXECUTION_PAYLOAD_VALUE))
+        .isEqualTo(executionPayloadValue.toDecimalString());
+    final BlindedBlockContents result =
+        (BlindedBlockContents)
+            spec.getGenesisSchemaDefinitions()
+                .getBlindedBlockContainerSchema()
+                .sszDeserialize(Bytes.of(response.body().bytes()));
+    assertThat(result).isEqualTo(blindedBlockContents);
+  }
```

Comment:
We may need to add error cases test to this, but I'd separate it

Reply:
Added some integration test for empty block production result, missing execution payload result in cache and missing execution payload value